charts/internal/openstack-infra/templates/main.tf

```diff
@@ -43,7 +43,7 @@ resource "openstack_networking_subnet_v2" "cluster" {
 
 resource "openstack_networking_router_interface_v2" "router_nodes" {
   router_id = "{{ required "router.id is required" $.Values.router.id }}"
```

Comment:
I think this PR is not fixing all of the usage of the old syntax. For example 
```
  router_id = "{{ required "router.id is required" $.Values.router.id }}"
```

can be 
```
  router_id = "${openstack_networking_router_v2.router.id}"
```

which is still deprecated usage.

Reply:
Good point! The `DefaultRouterID` should be fixed now by 
```
DefaultRouterID = "openstack_networking_router_v2.router.id"
```

Reply:
Hmm, still:
```
  router_id = "openstack_networking_router_v2.router.id"
```

!=


```
  router_id = openstack_networking_router_v2.router.id
```

We need the quotes only when and existing ID is specified. We don't need the quotes when we use the tf expresison.

Reply:
Actually we don't need the quotes in the rendered helm output at all.

Reply:
When routerID is specified, you can pass the chart value explicitly quoted. At least this was my approach in the respective PR for provider-aws.

Reply:
Not sure if I am missing something but this is the same approach as in your aws provider PR https://github.com/ialidzhikov/gardener-extension-provider-aws/blob/25b75a6ae4c08a850c04d68c6a8516fbbb6deb12/pkg/controller/infrastructure/actuator_reconcile.go#L111

Reply:
```
  router_id = "openstack_networking_router_v2.router.id"
```

should fail that there is no existing router with this id.